core/arch/arm/plat-versal/main.c

```diff
@@ -56,3 +66,35 @@ void console_init(void)
 		   CONSOLE_UART_CLK_IN_HZ, CONSOLE_BAUDRATE);
 	register_serial_console(&console_data.chip);
 }
+
+static TEE_Result platform_banner(void)
+{
+	vaddr_t plm_rtca = (vaddr_t)phys_to_virt(PLM_RTCA, MEM_AREA_IO_SEC,
+						 PLM_RTCA_LEN);
+	uint8_t version = 0;
+	bool ahwrot = false;
+	bool shwrot = false;
+
+	assert(plm_rtca);
+
+	if (versal_soc_version(&version)) {
+		EMSG("Failure to retrieve SoC version");
+		return TEE_ERROR_GENERIC;
+	}
+
+	IMSG("Platform Versal:\tSilicon Revision v%d", version);
+
+	if (io_read32(plm_rtca + VERSAL_AHWROT_REG) == VERSAL_AHWROT_SECURED)
+		ahwrot = true;
+
+	if (io_read32(plm_rtca + VERSAL_SHWROT_REG) == VERSAL_SHWROT_SECURED)
+		shwrot = true;
+
+	IMSG("Hardware Root of Trust:\t"
+	     "Asymmetric %s Enabled, Symmetric %s Enabled",
+	     ahwrot ? "" : "NOT", shwrot ? "" : "NOT");
```

Comment:
This is more complicated than necessary, you could assign the correct strings above instead.